### DISCUSSION TITLE
Instantiating the LLM with the AzureChatOpenAI class while using FalkorDB.

### DISCUSSION THREAD

**Discussion by pathan33** [FalkorDB/FalkorDB#792] [General]
Does FalkorDB support LLM instantiation with the AzureChatOpenAI class? If yes provide an example.

**Comment by AviAvni** [comment]
We dont support AzureChatOpenAI but from the documentation it is related to specific llm models like ChatGPT what are you trying to achieve?

**Reply by pathan33** [reply]
- We tried creating the ontologies by following the sample examples they have provided. - [LINK](https://github.com/FalkorDB/GraphRAG-SDK/?tab=readme-ov-file), but we are facing challenges while instantiating the LLM.

- Since we have the LLM subscriptions through the Azure OpenAI Service, we use the AzureChatOpenAI class with the Model Name, Deployment ID, Base URL, and API Key for instantiation. However, this approach is not allowing us to create the ontologies and KnowledgeGraphs.

- In the examples, for model instantiation "OpenAiGenerativeModel" is used and further this model is used in generating the ontology and KnowledgeGraphs.

**Reply by AviAvni** [reply]
Would like to do a meeting so we learn more about your use-case?
@galshubeli Will work on adding AzureChatOpenAI as a model to [FalkorDB/GraphRAG-SDK](https://github.com/FalkorDB/GraphRAG-SDK)
